File: Python/check_paranthesis.py
def is_balanced(s):
    stack = []
    for i in range(0, len(s)):
        c = s[i]
        if c == "(":
            stack.append(")")
        if c == "[":
            stack.append("]")
        if c == "{":
            stack.append("}")
        elif c == ")" or c == "]" or c == "}":
            if len(stack) == 0 or stack.pop() != c:
                return False
    
    return True


if __name__ == "__main__":
    s = "{[]]}"
    print(is_balanced(s))

File: Java/String/CheckParanthesis.java
package Java.String;

import java.util.Stack;

public class CheckParanthesis {
   public static boolean isBalanced(String str){
        
        Stack<Character> stack = new Stack<>();
        for (int i = 0; i <str.length() ; i++) {
            char c = str.charAt(i);
            if(c=='('){
                stack.push(')');
            }
            if(c=='['){
                stack.push(']');
            }
            if(c=='{'){
                stack.push('}');
            }else if (c==')' || c=='}' || c==']' ){
                if(stack.isEmpty() || stack.pop()!=c)
                    return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        String str = "({(}))";
        System.out.print(isBalanced(str));

    }
}
